get_s3.go

```diff
@@ -302,9 +299,18 @@ func (g *S3Getter) newS3Client(
 			return nil, err
 		}
 	} else {
-		config := g.getAWSConfig(region, url, creds)
+		config := g.getAWSConfig(url, creds)
 		sess = session.New(config)
 	}
 
+	// When using an AWS_PROFILE, the region is coming from your ~/.aws/config
+	// This may not exist in your profile, and it also may differ from the bucket
+	// location, so we need to override and explicitly bind what region we're
+	// intending based on the bucket location otherwise it's possible to get
+	// a MissingRegion error.
+	if region != "" {
```

Comment:
I didn't want to make this change since I didn't look _that_ deep, but I think we can avoid this check entirely since afaict, this value will always be something. Any detection of region that gets passed in here comes through at least with a fallback of `us-east-1` when there's no region defined in the url itself. But I kept it since it existed in `getAWSConfig` before.